### PR TITLE
Refactor Sentry loading

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Refactor Sentry loading to make the conditional checks clearer

To merge into https://github.com/guardian/dotcom-rendering/pull/6813

## Why?

The current logic and flow was difficult to reason about and made more complex by #6813

### CODE REVIEW

File: dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts
Comment (mxdvl) [inline dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts:30]: Should this be extracted to the decision to inject Sentry?

File: dotcom-rendering/src/web/browser/sentryLoader/init.ts
Comment (chrislomaxjones) [inline dotcom-rendering/src/web/browser/sentryLoader/init.ts:49]: Slightly more DCR-esque?

```suggestion
	const { switches, isDev, tests } = window.guardian.config;
	const enableSentryReporting = !!switches.enableSentryReporting;
	const isInBrowserVariantTest =
		BUILD_VARIANT && tests[dcrJavascriptBundle('Variant')] === 'variant';
```

Comment (arelra) [reply dotcom-rendering/src/web/browser/sentryLoader/init.ts:49]: That is nicer thanks!

File: dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts
Comment (arelra) [reply dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts:30]: I'm just wondering why this wasn't made part of the initial checks?

Is detecting an ad-blocker relatively '[expensive](https://github.com/guardian/commercial-core/blob/3b2a5f35275ef825fa3aa04d8a17a74618de374f/src/detect-ad-blocker.ts#L24)' and therefore we don't want it to block the init path?

`isAdBlockInUse` via  `useAdBlockInUse` is currently only used in islands which don't run on the init path.

Comment (arelra) [reply dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts:30]: It's a good point though. I'll add a comment for future travellers.

**Comment by bryophyta** [comment]
This looks good to me! Just thought I'd flag that I'm hoping we'll be able to add another condition that will bypass the random sampling, similar to the browser variant condition in this PR. (Issue #6198)

imo this issue should be addressed in a separate PR, and your refactor should make it easier to add these sorts of conditions, which is great! But thought it was worth flagging for visibility here

**Review by chrislomaxjones — APPROVED** [review]
Looks good to me! I'd probably wait on another approval from @guardian/dotcom-platform though :)

**Review by mxdvl — APPROVED** [review]
As a former member of @guardian/dotcom-platform, I find this 👌

**Pull request merged**
